app/Http/Controllers/AssetsController.php

```diff
@@ -1261,4 +1261,28 @@ public function auditStore(Request $request, $id)
             return redirect()->to("hardware")->with('success', trans('admin/hardware/message.audit.success'));
         }
     }
+
+    /**
+     * Generate labels for given asset ids.
+     *
+     * @return View
+     */
+    public function generateLabels(Request $request)
+    {
+        if (!$request->has('ids')) {
+            return redirect()->back()->with('error', 'No assets selected');
+        }
+
+        $asset_raw_array = $request->input('ids');
+        foreach ($asset_raw_array as $key => $asset_id) {
+            $asset_ids[] = $asset_id;
+        }
```

Comment:
I'm confused about what the foreach is doing here?  Is it just an array format issue that prevents asset_raw_array from working?